Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs

```diff
@@ -849,30 +849,39 @@ public virtual void LoadFromWagFile(string wagFilePath)
                 }
 
                 if (!FreightAnimations.MSTSFreightAnimEnabled) FreightShapeFileName = null;
-                    if (FreightAnimations.WagonEmptyWeight != -1)
+                if (FreightAnimations.WagonEmptyWeight != -1)
+                {
+                    // Computes mass when it carries containers
+                    float totalContainerMassKG = 0;
+                    if (FreightAnimations.Animations != null)
                     {
+                        foreach (var anim in FreightAnimations.Animations)
+                            if (anim is FreightAnimationDiscrete discreteAnim && discreteAnim.Container != null)
+                            {
+                                totalContainerMassKG += discreteAnim.Container.MassKG;
```

Comment:
Probably it would be cleaner if you use LINQ. However, this is a personal preference, so it's up to you.

Reply:
I have few knowledges about LINQ, so I prefer to leave it so.